cli/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# Build a Debian based image with grpc_cli as the entrypoint.
+#
+# Suppose you have a gRPC service running on your laptop at localhost:50051. Call it in
+# grpc_cli:
+# 
+# docker run -v `pwd`:/defs --rm -it test-grpc-cli call docker.for.mac.localhost:50051
+#   LinkShortener.ResolveShortLink "short_link:'asdf'" --protofiles=link_shortener.proto
+#
+# Note the address - since grpc_cli is running on the host machine, localhost won't work.
+# Docker for Mac provides the special docker.for.mac.localhost DNS address that resolves
+# to the host machine's IP.
+#
+# The user should mount their `pwd` to /defs.
+
+from debian AS build
+
+RUN apt-get update
+RUN apt-get install -y git libgflags-dev build-essential automake autoconf libtool shtool curl unzip
+WORKDIR /tmp
+RUN git clone https://github.com/grpc/grpc
+
+WORKDIR /tmp/grpc
+RUN git submodule update --init
+
+# Build grpc_cli
+# For some reason, we have to do this manually.
+WORKDIR /tmp/grpc/third_party/protobuf
+RUN ./autogen.sh
+
+WORKDIR /tmp/grpc
+RUN make grpc_cli
```

Comment:
Why not run this in https://github.com/namely/docker-protoc/blob/master/all/Dockerfile and just create a new dockerfile with protoc-all as base and a new entrypoint? You'd remove this file and duplication completely.